src/audio-manager.ts

```diff
@@ -1,261 +1,389 @@
+     * @throws If the given ID does not have a buffer associated with it, or all players are currently occupied.
      */
-    async play(posVec?: Float32Array): Promise<void> {
-        if (this._destroy) {
-            throw 'playable-node: play() was called on destroyed node!';
+    async play(id: number, config?: PlayConfig) {
+        const buffer = this._bufferCache.get(id);
+        if (!buffer) {
+            throw `audio-manager: No identifier with number: ${id} found!`;
         }
-        if (this._isPlaying) {
-            this.stop();
-        } else if (_audioContext.state === 'suspended') {
-            await _unlockAudioContext();
+        const player = this._busyPlayers.get(id) || this._freePlayers.pop();
+        if (!player) {
+            throw 'audio-manager: Reached maximum number of simultaneously playable sounds!';
         }
-        this._audioNode = new AudioBufferSourceNode(_audioContext, {
-            buffer: this._audioBuffer,
-            loop: this.loop,
-        });
-        if (posVec !== undefined) {
-            this._pannerNode = new PannerNode(_audioContext, {
-                coneInnerAngle: 360,
-                coneOuterAngle: 0,
-                coneOuterGain: 0,
-                distanceModel: 'exponential' as DistanceModelType,
-                maxDistance: 10000,
-                refDistance: 1.0,
-                rolloffFactor: 1.0,
-                panningModel: this.HRTF ? 'HRTF' : 'equalpower',
-                positionX: posVec![0],
-                positionY: posVec![2],
-                positionZ: -posVec![1],
-                orientationX: 0,
-                orientationY: 0,
-                orientationZ: 1,
-            });
-            this._audioNode.connect(this._pannerNode!).connect(this._gainNode);
-        } else {
-            this._audioNode.connect(this._gainNode);
+        player.stop();
+        this._busyPlayers.set(id, player);
+        if (_audioContext.state === 'suspended') {
+            await unlockAudioContext();
         }
-        this._audioNode.addEventListener('ended', this.stop);
-        this._audioNode.start();
-        this._isPlaying = true;
+        player.play(buffer, id, config);
+        this.emitter.notify({id: id, state: PlayState.PLAYING});
     }
 
     /**
-     * This is an alternative to the regular `play()` function, with advanced customization options for the distance
-     * model and directional fall-off.
+     * Plays the audio file associated with the given ID until it naturally ends.
+     *
+     * @note
+     * - IDs can be triggered as often as there are one-shot players in the AudioManager.
+     * - One shots work with First-In-First-Out principle. If all players are occupied, the manager will stop the
+     * one that started playing first, to free up a player for the new ID.
+     * //@todo: Question for Timmy: Not sure if SFX will suit most use cases or MASTER is the better choice.
```

Comment:
I think master and music are less suitable. SFX for oneshots is fine. You don't want to play anything directly on master. You can play music that is not loopable, but I don't thing it's ever a one shot.